sdk/keyvault/azsecrets/models.go

```diff
@@ -88,9 +88,13 @@ type Attributes struct {
 }
 
 func (s Attributes) toGenerated() *internal.SecretAttributes {
+	var recLevel *internal.DeletionRecoveryLevel
+	if s.RecoveryLevel != nil {
+		recLevel = s.RecoveryLevel.toGenerated().ToPtr()
```

Comment:
Not saying you have to do this, but you could simplify this a bit by having `toGenerated()` take its receiver by pointer and perform the `nil` check there (it's legal to call a method on a `nil` thing as long as you pass by pointer and check for `nil`).  Then you just do `RecoveryLevel: s.RecoveryLevel.toGenerated()`.